source/elements/oneTBB/source/algorithms/blocked_ranges/blocked_rangeNd_cls.rst

```diff
@@ -0,0 +1,171 @@
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+===============
+blocked_rangeNd
+===============
+**[algorithms.blocked_rangeNd]**
+
+Class template that represents a recursively divisible N-dimensional half-open interval.
+
+A ``blocked_rangeNd`` is the N-dimensional extension of ``blocked_range``, ``blocked_range2d`` and ``blocked_range3d``.
+We can interpret it as a Cartesian product of N instances of ``blocked_range``.
+Different to ``blocked_range2d`` and ``blocked_range3d``, all ranges have to be specified over the same type ``Value``.
+Therefore, ``blocked_rangeNd<int,2>`` is a specialisation of ``blocked_range2d`` where both dimensions are of type ``int``.
+
+.. code:: cpp
+
+    namespace oneapi {
+        namespace tbb {
+
+            template<typename Value, unsigned int N>
+            class blocked_rangeNd {
+            public:
+                // Types
+                using value_type = Value;
+                using dim_range_type = blocked_range<value_type>;
+                using size_type = typename dim_range_type::size_type;
+
+                // Constructors
+                blocked_rangeNd(const dim_range_type& dim0 /*, ... - exactly N parameters of the same type*/);
+                blocked_rangeNd(value_type size[N], size_type grainsize = 1);
+                blocked_rangeNd(blocked_rangeNd& r, split); 
+                blocked_rangeNd(blocked_rangeNd& r, proportional_split proportion); 
+
+                // Capacity
+                static constexpr unsigned int dim_count();
+                bool empty() const;
+
+                // Access
+                bool is_divisible() const;
+                const dim_range_type& dim(unsigned int dimension) const;
+            };
+
+        } // namespace tbb
+    } // namespace oneapi        
+
+Requirements:
+
+* The *Value* must meet the :doc:`BlockedRangeValue requirements <../../named_requirements/algorithms/blocked_range_val>`
+
+Member types
+------------
+
+.. code:: cpp
+
+    using value_type = Value;
+
+The type of the range values.
+
+.. code:: cpp
+
+    using dim_range_type = blocked_range<value_type>;
+
+The type that represents one out of the N dimensions.
+
+.. code:: cpp
+
+    using size_type = typename dim_range_type::size_type;
+
+The type for measuring the size of a dimension.
+
+Member functions
+----------------
+
+.. code:: cpp
+
+    blocked_rangeNd( const dim_range_type& dim0 /*, ... - exactly N parameters of the same type*/ );
+
+**Effects:**  Constructs a ``blocked_rangeNd`` representing an N-dimensional space of values.
+The space is the half-open Cartesian product of ranges ``dim0 x ...`` each having its own grain size.
+The constructor must take exactly N arguments which types match to ``const dim_range_type&``.
+
+**Example:** For ``blocked_rangeNd<int,4>``, this constructor is equivalent to
+``blocked_rangeNd( const blocked_range<int>&, const blocked_range<int>&, const blocked_range<int>&, const blocked_range<int>& )``.
+
+.. note::
```

Comment:
Maybe rephrase to smth like it for an easier understanding: 

A variadic template constructor, template <typename... Dims> blocked_rangeNd(const Dims&... dims), does not work correctly in all cases. Even if you set specific size and type rules for the Dims parameter pack, it does not fully comply because the types in Dims cannot be automatically determined when using braced initialization lists. As a result, expressions like blocked_rangeNd<int, 4>{{0,1},{0,2},{0,3},{0,4}} cause a compilation error.

Reply:
This note is for class implementors, not for its users. It is intended to describe a potential pitfall of a seemingly equivalent but in fact subtly different form of the class constructor, which therefore should not be used by implementors as a substitution for the constructor specified above the note. This is why I use Subjunctive mood in the note instead of simpler verb forms.

The use of a note (and not a plain text) is also intentional, as I wanted to indicate that this part is informational, not normative (see more details at https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:nonnormativerefs). Perhaps this approach is not used consistently in the specification, but I find it applicable and quite useful here. I guess this note can be changed to a caution - I see some uses of it in the oneTBB spec.

Reply:
Reworked it somewhat, hopefully for better.